src/storage/UpdateEdgeProcessor.cpp

```diff
@@ -0,0 +1,373 @@
+        CHECK_EQ(returnColumnsNum_, writeSchema->getNumFields());
+        auto reader = RowReader::getRowReader(encoded, writeSchema);
+        int64_t index = 0;
+        for (auto& prop : this->edgeContext_.props_) {
+            if (prop.returned_) {
+                auto res = RowReader::getPropByIndex(reader.get(), index);
+                if (!ok(res)) {
+                    VLOG(3) << "Bad value for prop " << prop.prop_.get_name();
+                    return kvstore::ResultCode::ERR_FALSE_UPDATER;
+                }
+                record_[prop.retIndex_] = value(std::move(res));
+                index++;
+            }
+        }
+        for (auto& tc : tagContexts_) {
+            for (auto& prop : tc.props_) {
+                if (prop.returned_) {
+                    auto res = RowReader::getPropByIndex(reader.get(), index);
+                    if (!ok(res)) {
+                        VLOG(3) << "Bad value for prop " << prop.prop_.get_name();
+                        return kvstore::ResultCode::ERR_FALSE_UPDATER;
+                    }
+                    record_[prop.retIndex_] = value(std::move(res));
+                    index++;
+                }
+            }
+        }
+    }
+
+    // the Set clause for updating
+    SchemaVer ver = edgeReader_->schemaVer();
+    const auto constSchema = this->schemaMan_->getEdgeSchema(
+            spaceId_, this->edgeContext_.edgeType_, ver);
+    auto schema = std::const_pointer_cast<meta::SchemaProviderIf>(constSchema);
+    // TODO(zhangguoqing) When the edge is to be inserted, it can't instantiate
+    // the class RowUpdater() with edgeReader_ parameter, otherwise the updater_
+    // doing encoded will report the error: Failed to encode updated data.
+    if (resp_.get_upsert()) {
+        updater_ = std::unique_ptr<RowUpdater>(new RowUpdater(schema));
+    } else {
+        updater_ = std::unique_ptr<RowUpdater>(new RowUpdater(std::move(edgeReader_), schema));
+    }
+    for (auto& item : updateItems_) {
+        auto edgeName = item.get_name();
+        auto field = item.get_field();
+        StatusOr<std::unique_ptr<Expression>> exp = Expression::decode(item.get_value());
+        auto vexp = std::move(exp).value();
+        vexp->setContext(expCtx_.get());
+        auto value = vexp->eval();
+        if (!value.ok()) {
+            return kvstore::ResultCode::ERR_FALSE_UPDATER;
+        }
+        auto expValue = value.value();
+        // update the props and the record
+        int32_t retIndex = -1;
+        if (returnColumnsNum_ > 0) {
+            for (auto& prop : this->edgeContext_.props_) {
+                if (prop.prop_.name == field && prop.returned_) {
+                    retIndex = prop.retIndex_;
+                    break;
+                }
+            }
+        }
+        switch (expValue.which()) {
+            case VAR_INT64: {
+                auto v = boost::get<int64_t>(expValue);
+                updater_->setInt(field, v);
```

Comment:
What if the filed is not int type?  how to deal with it

Reply:
The UpdateEdgeExecutor::prepareSet has checked the field in it's schema in graph layer.